packages/codefixes/src/fixes/addMissingTypesBasedOnInheritance/index.ts

```diff
@@ -0,0 +1,60 @@
+import { findNodeAtPosition } from '@rehearsal/utils';
+import { findAncestor, isClassDeclaration, isMethodDeclaration, isParameter } from 'typescript';
+import { createCodeFixAction } from '../../hints-codefix-collection';
+import type { CodeFixAction } from 'typescript';
+import type { CodeFix, DiagnosticWithContext } from '../../types';
+
+export class AddMissingTypesBasedOnInheritanceCodeFix implements CodeFix {
+  getCodeAction(diagnostic: DiagnosticWithContext): CodeFixAction | undefined {
+    const node = findNodeAtPosition(diagnostic.file, diagnostic.start, diagnostic.length);
+
+    // Support only function params
+    if (!node || !isParameter(node)) {
+      return undefined;
+    }
+
+    const currentMethod = findAncestor(node, isMethodDeclaration);
+    const currentClass = findAncestor(node, isClassDeclaration);
+
+    // Method declaration inside class extends another class
+    if (!currentMethod || !currentClass || !currentClass.heritageClauses) {
+      return undefined;
+    }
+
+    const checker = diagnostic.checker;
+
+    // One way to check ancestor's method param type... doesn't work :)
+    /*
+    for (const heritageClause of currentClass.heritageClauses) {
+      for (const heritageClauseTypeNode of heritageClause.types) {
+        const baseType = checker.getTypeAtLocation(heritageClauseTypeNode);
+
+        const ancestorMethod = baseType.getProperty(currentMethod.name.getText());
+
+        if (!ancestorMethod) {
+          continue;
+        }
+
+        const resolvedType = checker.getTypeOfSymbolAtLocation(
+          ancestorMethod,
+          heritageClauseTypeNode
+        );
+
+        // Expect this to return type, but `any` is returner
+        // checker.typeToString(checker.getTypeOfSymbolAtLocation(ancestorMethod.getDeclarations()[0].parameters[0].symbol.valueDeclaration))
```

Comment:
Expect to get type here, but having `any`. 

The `ancestorMethod` is part of `Animal`